File: Android 10 - Android services, Work managers - executing long running tasks/ServiceExercise_2/app/src/main/java/com/programming/android/sdu/serviceexercise/NextActivity.java
package com.programming.android.sdu.serviceexercise;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;


public class NextActivity extends BaseActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.i("service_exrecise", "NextActivity onCreate- Current Thread ID- " + Thread.currentThread().getId() + " For Thread- " + Thread.currentThread().getName());

        setContentView(R.layout.next_activity);
        initViews();
        textSwitcher.setInAnimation(this, android.R.anim.slide_in_left);
        textSwitcher.setOutAnimation(this, android.R.anim.slide_out_right);

        findViewById(R.id.btnBack).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }

}

File: Android 10 - Android services, Work managers - executing long running tasks/ServiceExercise_2/app/src/main/java/com/programming/android/sdu/serviceexercise/NextActivity.kt
package com.programming.android.sdu.serviceexercise

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button

class NextActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.i("service_exrecise", "NextActivity onCreate- Current Thread ID- " + Thread.currentThread().id + " For Thread- " + Thread.currentThread().name)
        setContentView(R.layout.next_activity)
        initViews()
        textSwitcher.setInAnimation(this, android.R.anim.slide_in_left)
        textSwitcher.setOutAnimation(this, android.R.anim.slide_out_right)
        val btnBack: Button = findViewById(R.id.btnBack)
        btnBack.setOnClickListener { finish() }
    }

}